clients/bigquery/cast.go

```diff
@@ -59,11 +60,23 @@ func castColVal(colVal any, colKind columns.Column, additionalDateFmts []string)
 			return fmt.Sprintf(`{"key":"%s"}`, constants.ToastUnavailableValuePlaceholder), nil
 		}
 
-		colValString, isOk := colVal.(string)
-		if isOk && colValString == "" {
-			// Empty string is not a valid JSON object, so let's return nil.
-			return nil, nil
+		// Structs from relational and Mongo are different.
+		// MongoDB will return the native objects back such as `map[string]any{"hello": "world"}`
+		// Relational will return a string representation of the struct such as `{"hello": "world"}`
+		if colValString, isOk := colVal.(string); isOk {
+			if colValString == "" {
+				return nil, nil
+			}
+
```

Comment:
Would it make sense to move the check above for the `constants.ToastUnavailableValuePlaceholder` here? Otherwise we're potentially calling `fmt.Sprint()` on a map.

Reply:
Even if we did call `fmt.Sprint(...)` on a map, it'd be fine? We're really just checking to make sure the value doesn't contain `__debezium_unavailable_value`

Reply:
Yeah it'd still work, I was mostly thinking that in the case that `colVal` is a large nested data structure it would be inefficient to convert it to a string.